### PR TITLE
feat(summarization): pass generatePrompts to Mystique for Impact Engine

### PR DESCRIPTION
**Pull request by anuj-adobe**
## Summary

Forward the `generatePrompts` kwarg from Slack/API producers through the summarization audit chain so it reaches the outbound `guidance:summarization` Mystique payload — unblocking **Impact Engine** prompt generation for summarization suggestions (same hook that prerender already uses).

- Linked mystique PR: https://git.corp.adobe.com/experience-platform/mystique/pull/2578

## What changed

`src/summarization/handler.js`:
- Private helper `parseGeneratePromptsFlag(data, log)` — reads a JSON string or object, warn-and-default-to-false on parse error.
- `importTopPages` (step 1): destructures `data` from context, calls the helper, returns `auditContext: { generatePrompts }` on all three return paths (empty SEO, success, catch).
- `submitForScraping` (step 2): forwards `generatePrompts: !!auditContext?.generatePrompts` alongside the existing `summarizationUrls` in the returned `auditContext`.
- `sendToMystique` (step 3): reads `const generatePrompts = !!auditContext?.generatePrompts;` and includes it in `message.data.generatePrompts` on the outbound Mystique SQS message. Mystique consumes this flag to decide whether to generate Impact Engine prompts alongside the summarization suggestions.

## Test plan

- [x] `src/summarization/handler.js` at **100% lines / branches / functions / statements**
- [x] `test/audits/summarization/handler.test.js`: **51 tests passing**, including 6 new cases covering:
  - parse from object data
  - parse from JSON string data
  - invalid JSON falls back to `false` and emits a warn log
  - `generatePrompts=false` propagated when SEO is empty
  - step-2 forwards `generatePrompts=true` from incoming auditContext
  - step-3 propagates `generatePrompts=true` into the Mystique payload
- [x] `npm run lint` clean
- [ ] Manual: trigger a summarization run from Slack with `generatePrompts: true` and confirm Mystique receives `data.generatePrompts === true` on the `guidance:summarization` message, and that Impact Engine prompts are generated downstream

### CODE REVIEW

File: src/common/step-audit.js [hunk 1]
```diff
@@ -164,6 +167,16 @@ export class StepAudit extends BaseAudit {
       const stepName = auditContext.next || stepNames[0];
       const isLastStep = stepName === stepNames[stepNames.length - 1];
       const step = this.getStep(stepName);
+
+      // Step-1 only: parse `message.data` into `auditContext.messageData`.
+      // Subsequent steps carry it forward via `preserveMessageData` in chainStep.
+      if (!hasNext) {
+        const messageData = parseMessageDataForRunnerAudit(data);
+        if (messageData !== undefined) {
+          auditContext.messageData = messageData;
```
Comment (MysticatBot) [inline src/common/step-audit.js]: issue (blocking): The CLAUDE.md documentation requirements state: "**Significant new functionality** must be developed spec-first. Add a spec to `/docs/specs/`." and "**Technical or architectural decisions with long-term impact** must be recorded as ADRs in `/docs/decisions/`."

This PR introduces a framework-level mechanism that changes how all audit handlers access SQS message parameters - a generic `messageData` passthrough that replaces per-handler bespoke parsing. This is both significant new functionality and an architectural decision with long-term impact (it establishes the pattern all future audits will follow).

Fix: Add a short spec to `docs/specs/` covering the problem, the design, and migration guidance. Add an ADR to `docs/decisions/` recording the decision to adopt framework-level messageData passthrough over per-audit parsing.

File: src/common/step-audit.js [hunk 2]
```diff
@@ -164,6 +167,16 @@ export class StepAudit extends BaseAudit {
       const stepName = auditContext.next || stepNames[0];
       const isLastStep = stepName === stepNames[stepNames.length - 1];
       const step = this.getStep(stepName);
+
+      // Step-1 only: parse `message.data` into `auditContext.messageData`.
+      // Subsequent steps carry it forward via `preserveMessageData` in chainStep.
+      if (!hasNext) {
+        const messageData = parseMessageDataForRunnerAudit(data);
```
Comment (MysticatBot) [inline src/common/step-audit.js]: issue (blocking): `parseMessageDataForRunnerAudit(data)` stores the entire parsed SQS `data` object as `messageData`. This means `messageData` carries routing/identity fields (`mode`, `opportunityId`, `scrapeJobId`) alongside the intended pass-through kwargs (`generatePrompts`). The PR description frames this as "keyword args flow from the step-1 SQS message," but the implementation passes everything.

This creates a misleading mental model: a future developer reading `auditContext.messageData.generatePrompts` might assume the object only contains forwarded kwargs, then accidentally consume raw routing fields as pass-through data.

Fix: Either (a) add a JSDoc comment on the `auditContext.messageData` assignment clarifying that it contains the full parsed `data` envelope (not just kwargs), or (b) filter to only recognized pass-through keys with a small allowlist. Option (a) is the minimal fix for this PR.

**Review by MysticatBot — CHANGES_REQUESTED** [review]
Hey @anuj-adobe,

**Verdict: Request changes** - solid framework design with well-tested mechanics, but the repo's documentation requirements need addressing.
**Complexity: MEDIUM** - medium diff touching framework internals across multiple audit paths.
_Changes: Centralizes SQS message data parsing into a framework-level `auditContext.messageData` passthrough, removing bespoke per-handler parsing (12 files)._

## Must fix before merge

1. [Important] Missing spec and ADR for significant new framework functionality - `docs/specs/`, `docs/decisions/` (details inline)
2. [Important] Full SQS `data` payload stored as `messageData` without filtering or documentation - `src/common/step-audit.js:174` (details inline)

<details>
<summary>Non-blocking (3): minor issues and suggestions</summary>

- nit: `preserveMessageData` accepts arrays (passes `typeof === 'object'` check) unlike `parseMessageDataForRunnerAudit` which rejects them - add `&& !Array.isArray(messageData)` for consistency - `src/common/audit-utils.js:140`
- suggestion: shallow-copy the inner `messageData` object in `preserveMessageData` to prevent cross-step mutation via shared reference (`{ messageData: { ...messageData } }`) - `src/common/audit-utils.js:140`
- nit: `AsyncJobRunner.chainStep` calls `preserveMessageData` but `AsyncJobRunner.run` never seeds `messageData` (unlike `StepAudit.run`) - a brief comment explaining this asymmetry would help future readers - `src/common/async-job-runner.js:53`
</details>

---

<sub><b>Skill:</b> pr-review | <b>Model:</b> us.anthropic.claude-opus-4-6-v1[1m] | <b>Duration:</b> 7m 37s | <b>Cost:</b> $7.55 | <b>Commit:</b> <code>a970c50f6f4e7e07aeb941b6a1d111779d64266a</code><br>If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>

**Review by MysticatBot — APPROVED** [review]
Hey @anuj-adobe,

**Verdict: Approve** - prior blocking findings addressed; framework design is sound and well-documented.
**Complexity: HIGH** - large diff (1100 lines, 16 files).
_Changes: Centralizes SQS message data parsing into a framework-level `auditContext.messageData` passthrough, removing bespoke per-handler parsing, and adds a comprehensive spec document (16 files)._
Note: Recommend a human read before merge - this change adds a spec to `docs/specs/` (architectural-significance path). The bot review is a complement to, not a replacement for, a human read here.

## Previously flagged, now resolved

- Spec requirement satisfied: `docs/specs/2026-06-25-generic-message-data-passthrough.md` added with problem statement, technical design, alternatives, migration guidance, and success criteria.
- Full-envelope design now documented: JSDoc on `preserveMessageData`, inline comments in `step-audit.js`, and spec section "What messageData contains" explain the rationale.
- Array rejection consistency: `preserveMessageData` includes `!Array.isArray(messageData)` guard.
- AsyncJobRunner asymmetry: comment block at lines 95-100 explains why `run` does not seed messageData.

<details>
<summary>Non-blocking (4): minor issues and suggestions</summary>

- nit: CLAUDE.md requires ADRs in `/docs/decisions/` for architectural decisions - technically unmet since the decision lives in the spec's "Alternatives Considered" section rather than a separate ADR file. The substance is documented; only the prescribed location differs. Consider adding a minimal ADR cross-referencing the spec, or clarifying in CLAUDE.md that specs with comprehensive alternatives sections satisfy the ADR requirement. - `docs/decisions/`
- nit: No unit assertion for the array-rejection branch of `preserveMessageData` (e.g. `preserveMessageData({ messageData: [1,2] })` returning `{}`). The code path is exercised by parser tests, but an explicit unit assertion documents the guard's intent. - `test/common/audit-utils.test.js`
- suggestion: Shallow-copy `messageData` in `preserveMessageData` (`{ messageData: { ...messageData } }`) to prevent accidental cross-step mutation via shared reference. SQS serialization isolates between steps, but within a single step execution the reference is shared. - `src/common/audit-utils.js:148`
- suggestion: `ai-only-handler.js` still manually parses `data` for `opportunityId`/`scrapeJobId` while reading `generatePrompts` from `messageData`. Consider migrating those reads to `messageData` in a follow-up for consistency, or add a TODO comment explaining the asymmetry. - `src/prerender/ai-only-handler.js:50`
</details>

---

<sub><b>Skill:</b> pr-review | <b>Model:</b> us.anthropic.claude-opus-4-6-v1[1m] | <b>Duration:</b> 2m 9s | <b>Cost:</b> $5.11 | <b>Commit:</b> <code>784921bc8c9e838101bf09a4d46577acb67fee45</code><br>If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>

**Review by MysticatBot — APPROVED** [review]
Hey @anuj-adobe,

> ⚠ Degraded review - no spec document was found for this change (searched the PR links, the touched repo's docs, the architecture/guidelines docs, and linked Jira). This review covers code-level quality but could not validate the change against an agreed design, so confidence is reduced. Add a spec link (PR template section 4) and re-request review for a full-confidence pass.

**Verdict: Approve** - clean, well-scoped change following the established prerender pattern with thorough test coverage.
**Complexity: MEDIUM** - medium diff, single service, per-audit scope.
_Changes: Threads a `generatePrompts` boolean through the summarization audit's 3-step pipeline (importTopPages, submitForScraping, sendToMystique) to reach the outbound Mystique SQS message (2 files)._

<details>
<summary>Non-blocking (2): minor issues and suggestions</summary>

- suggestion: Spread incoming `auditContext` in `submitForScraping` rather than constructing a fresh object (`{ ...auditContext, [AUDIT_CONTEXT_URLS_KEY]: topPagesToScrape, generatePrompts: ... }`) to avoid silently dropping future keys added by upstream steps - `src/summarization/handler.js:210`
- nit: Add a test for `data=undefined`/`data=null` input to `importTopPages` to document the graceful-default-false behavior of `parseGeneratePromptsFlag` - `test/audits/summarization/handler.test.js`
</details>

## Previously flagged, now resolved

- Framework-level `messageData` passthrough dropped per reviewer feedback; replaced with focused per-audit boolean threading.
- Spec/ADR requirement no longer applies (no framework changes in this revision).
- Full-envelope design concern resolved (only a single boolean is threaded, not the full SQS data payload).

---

<sub><b>Skill:</b> pr-review | <b>Model:</b> us.anthropic.claude-opus-4-6-v1[1m] | <b>Duration:</b> 2m 14s | <b>Cost:</b> $3.42 | <b>Commit:</b> <code>996fb90cb5ea2ec8264646d34dd30b83d928fe91</code><br>If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>

**Review by Danialkafeel — APPROVED** [review]
